### PR TITLE
Label cgroup driver for k8s and fix deployment with floating ip disabled for swarm mode

### PR DESCRIPTION
**Pull request by brtknr**

### CODE REVIEW

**Comment by markgoddard** [comment]
What's going on with the stackhpc/queens branch? Looks like we've lost our local changes.

**Comment by brtknr** [comment]
I renamed it to stackhpc/queens-old as I wanted to test and see if some of the earlier changes were necessary... specially those concerned with checking for Docker storage setup.

File: magnum/drivers/common/templates/fragments/configure_docker_storage_driver_atomic.sh [hunk 1]
```diff
@@ -47,5 +47,5 @@ configure_devicemapper () {
         echo "DATA_SIZE=95%FREE" >> /etc/sysconfig/docker-storage-setup
     fi
 
-    docker-storage-setup
+    container-storage-setup
```
Comment (markgoddard) [inline magnum/drivers/common/templates/fragments/configure_docker_storage_driver_atomic.sh]: This is certainly simpler. Have you checked that docker-ce actually uses the config file generated by this command? Looks like by default it writes to /etc/sysconfig/docker-storage. (See https://github.com/projectatomic/container-storage-setup).

File: magnum/drivers/common/templates/swarm/fragments/add-docker-daemon-options.sh [hunk 2]
```diff
@@ -2,12 +2,39 @@
...
-    opts=$opts"--tlskey=/etc/docker/server.key "
-    opts=$opts"--tlscert=/etc/docker/server.crt "
+if [[ -f /etc/sysconfig/docker ]]; then
+    SET_LOG_DRIVER=False
+else
+    SET_LOG_DRIVER=True
+    LOG_DRIVER=journald
+    # The Docker CE distribution does not provide sufficient SELinux support.
+    # With SELinux enabled, non-privilged containers are unable to (for
+    # example) bind to privileged ports as root in the container.
```
Comment (markgoddard) [inline magnum/drivers/common/templates/swarm/fragments/add-docker-daemon-options.sh]: This comment is now a bit lost.

**Comment by brtknr** [comment]
I suppose I could have called the new branch stackhpc/queens-new instead.

File: magnum/drivers/common/templates/swarm/fragments/add-docker-daemon-options.sh (hunk 2 above)
Comment (brtknr) [reply magnum/drivers/common/templates/swarm/fragments/add-docker-daemon-options.sh]: Well spotted!

File: magnum/drivers/common/templates/fragments/configure_docker_storage_driver_atomic.sh (hunk 1 above)
Comment (brtknr) [reply magnum/drivers/common/templates/fragments/configure_docker_storage_driver_atomic.sh]: Yup, it does!

File: magnum/drivers/common/templates/kubernetes/fragments/configure-kubernetes-master.sh [hunk 3]
```diff
@@ -140,7 +140,22 @@ EOF
...
+    KUBELET_ARGS="${KUBELET_ARGS} --cgroup-driver=${CGROUP_DRIVER}"
+
+    systemctl disable docker
+    if cat /usr/lib/systemd/system/docker.service | grep 'native.cgroupdriver'; then
+            cp /usr/lib/systemd/system/docker.service /etc/systemd/system/
+            sed -i "s/\(native.cgroupdriver=\)\w\+/\1$CGROUP_DRIVER/" \
+                    /etc/systemd/system/docker.service
+    else
+            cat > /etc/systemd/system/docker.service.d/cgroupdriver.conf << EOF
+ExecStart=---exec-opt native.cgroupdriver=$CGROUP_DRIVER
```
Comment (markgoddard) [inline magnum/drivers/common/templates/kubernetes/fragments/configure-kubernetes-master.sh]: There are three hyphens here, is that intentional?

File: magnum/drivers/common/templates/kubernetes/fragments/configure-kubernetes-minion.sh [hunk 4]
```diff
@@ -136,8 +136,25 @@ if [ -n "${INSECURE_REGISTRY_URL}" ]; then
...
+KUBELET_ARGS="${KUBELET_ARGS} --cgroup-driver=${CGROUP_DRIVER}"
+
+systemctl disable docker
+if cat /usr/lib/systemd/system/docker.service | grep 'native.cgroupdriver'; then
+        cp /usr/lib/systemd/system/docker.service /etc/systemd/system/
+        sed -i "s/\(native.cgroupdriver=\)\w\+/\1$CGROUP_DRIVER/" \
+                /etc/systemd/system/docker.service
+else
+        cat > /etc/systemd/system/docker.service.d/cgroupdriver.conf << EOF
+ExecStart=---exec-opt native.cgroupdriver=$CGROUP_DRIVER
```
Comment (markgoddard) [inline magnum/drivers/common/templates/kubernetes/fragments/configure-kubernetes-minion.sh]: As above.

**Pull request merged**